crates/module-system/sov-solana-offchain-auth/src/authentication.rs

```diff
@@ -0,0 +1,483 @@
+use std::fmt::Debug;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use sov_modules_api::capabilities::{
+    calculate_hash_metered, extract_authorization_data, verify_chain_id, AuthenticationError,
+    AuthenticationOutput, FatalError, UniquenessData,
+};
+use sov_modules_api::macros::UniversalWallet;
+use sov_modules_api::transaction::{
+    self, AuthenticatedTransactionAndRawHash, TransactionCallable, TxDetails, UnsignedTransaction,
+};
+use sov_modules_api::{
+    CryptoSpec, DispatchCall, GasMeter, GasSpec, MeteredJsonDeserialize,
+    MeteredJsonDeserializeError, MeteredSignature, ProvableStateReader, Spec, TxHash,
+};
+
+/// The application domain for Solana offchain messages (placeholder for now)
+pub const APPLICATION_DOMAIN: [u8; 32] = [0u8; 32];
+
+/// The payload for a solana offchain message.
+/// Essentially a wrapper around `sov_modules_api::transaction::UnsignedTransaction` that also
+/// includes the chain_hash, in order to ensure the hash gets signed as part of the message.
+/// We duplicate the UnsignedTransaction type rather than wrapping it to ensure the JSON displayed
+/// to the user doesn't get too nested.
+#[serde_with::serde_as]
+#[derive(Debug, Serialize, Deserialize, UniversalWallet)]
+#[serde(bound = "R::Call: serde::Serialize + serde::de::DeserializeOwned")]
+pub struct SolanaOffchainUnsignedTransaction<R: TransactionCallable, S: Spec> {
+    /// The runtime call
+    pub runtime_call: R::Call,
+    /// The uniqueness identifier
+    pub uniqueness: UniquenessData,
+    /// Data related to fees and gas handling.
+    pub details: TxDetails<S>,
+    /// The chain hash the transaction data was signed with. Because we're not actually using the
+    /// schema to display the transaction, we have to include and validate the hash explicitly.
+    #[serde_as(as = "serde_with::hex::Hex")]
+    pub chain_hash: [u8; 32],
+}
```

Comment:
Had we decided that this chain_hash was no longer required?

Reply:
Sorry for the confusion - those changes are still in #1526! I don't want to merge now since you've already reviewed this - on re-review, feel free to press merge there and just re-review the combined branch, or have it as a follow-up.

Reply:
Merged that PR in for cleanup - this should leave only one PR to review.